### ISSUE TITLE
unify our two caching mechanisms

### ISSUE THREAD

**Issue by Pomax** [mozilla/webmaker-core#591]
we're using two mechanisms for caching atm,
1. mem storage (used for params, shared prefs)
2. java caching in the "java API" (used for caching data between view changes)

Having both is a little dev-unfriendly, so it'll be worth unifying these into a single caching solution.

**Comment by ryanwarsaw** [comment]
@Pomax Shouldn't this have the android tag?

Edit: I'd love to take this ticket.

**Comment by Pomax** [comment]
it should (thanks @xmatthewx!) and if you want to tackle this one: have at it! I can't assign you (probably a permissions issue?) but consider it yours =)

**Comment by ryanwarsaw** [comment]
@Pomax Yeah, you can't assign me to a ticket because I'm not apart of the organization, but there is a assigned to contributor tag I use to keep track of the tickets I'm assigned to and keep everything organized.

**Comment by Pomax** [comment]
cool, in that case: have an assigned-to-contributor label, good sir!

**Comment by ryanwarsaw** [comment]
I opened a PR that migrates gets rid of the old cache and migrates to the new one, with the old API calls. I'll work over the weekend on moving these API calls to the new setup and then eventually remove the old ones once it has landed.

**Comment by xmatthewx** [comment]
Let's make a QA checklist for testing this on device.

**Comment by ryanw-se** [comment]
I'd like to finish this off this week, I attempted to do this twice but for some reason I was getting funky results when clearing some from the Queue (It was telling me stuff that existed, didn't exist), but still worked when you cleared all payloads for a key.
